### PR TITLE
perf: change return value to NodeData to determine which DAG it belongs.

### PR DESCRIPTION
**Pull request by kobayu858**
- Changed the return value to NodeData to make it easier to see which DAG it belongs to.
- PR for #26

### CODE REVIEW

File: lib/src/graph_extension.rs [hunk 1]
```diff
@@ -479,6 +480,12 @@ impl GraphExtension for Graph<NodeData, i32> {
         }
     }
 
+    fn set_dag_id(&mut self, dag_id: usize) {
+        for node_i in self.node_indices() {
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:484]: Add error handling when the graph has no nodes.

File: lib/src/graph_extension.rs [hunk 2]
```diff
@@ -1061,6 +1068,16 @@ mod tests {
         assert_eq!(dag.get_parallel_process_nodes(n0), None);
     }
 
+    #[test]
+    fn test_set_dag_id_normal() {
+        let mut dag = Graph::<NodeData, i32>::new();
+        dag.set_dag_id(0);
+
+        for node_i in dag.node_indices() {
+            assert_eq!(dag[node_i].params["dag_id"], 0);
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:1077]: This line is not executed because dag has no nodes.

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/graph_extension.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:492]: Changed to cause panic when the number of nodes is 0.

File: lib/src/graph_extension.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:1101]: Add two nodes.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Pull request merged**